app/jobs/v3/services/update_broker_job.rb

```diff
@@ -65,7 +68,7 @@ def perform
           @warnings
         rescue => e
           begin
-            broker.update(state: previous_broker_state)
+            initial_broker.update(state: previous_broker_state)
```

Comment:
First I thought that `broker.reload.update...` could be an alternative to using `dup`; but this would result in an additional `SELECT`.

I think the best way to set the previous state would be:
```
ServiceBroker.where(id: update_request.service_broker_id).update(state: previous_broker_state)
```

Reply:
Jep, I also thought about broker.reload first, but wanted to prevent the additional select. Indeed the combined statement is better than using dup.